### ISSUE TITLE
3.0.0-Beta 3: Custom notification upgrade guide code doesn't work.

### ISSUE THREAD

**Issue by WiXSL** [marmelab/react-admin#3894]
The code proposed here [UPGRADE.md](https://github.com/marmelab/react-admin/blob/next/UPGRADE.md#custom-notification-components-must-emit-undoevents) for using custom notifications doesn't seem to be working.

I've added a custom notification for the demo app with the upgrade guide code here:
[https://codesandbox.io/s/react-admin-demo-300-gofzw](https://codesandbox.io/s/react-admin-demo-300-gofzw)

**Comment by WiXSL** [comment]
My mistake! Clossing

**Comment by WiXSL** [comment]
Now i've added styles to avoid the errors, but the notification won't show.

**Comment by fzaninotto** [comment]
I think the UPGRADE snippet misses the end part, which shows that you must connect the component to the Redux store:

```jsx
import { connect } from 'react-redux';
import {
    complete,
    undo,
    getNotification,
    hideNotification,
    Button,
    undoableEventEmitter
} from 'react-admin';
import compose from 'recompose/compose';

...

const mapStateToProps = state => ({
    notification: getNotification(state),
});

export default compose(
    withStyles(styles),
    connect(
        mapStateToProps,
        {
            complete,
            hideNotification,
            undo,
        }
    )
)(Notification);
```

Can you confirm that it solves the problem?

**Comment by WiXSL** [comment]
Ok, now is working. I've updated the PR.

**Comment by fzaninotto** [comment]
Fixed by #3893
